library/network_connections.py

```diff
@@ -1052,8 +1054,15 @@ def connection_create(self, connections, idx, connection_current=None):
             for nameserver in ip["dns"]:
                 if nameserver["family"] == socket.AF_INET:
                     s_ip4.add_dns(nameserver["address"])
-            for search_domain in ip["dns_search"]:
-                s_ip4.add_dns_search(search_domain)
+            # DNS (the protocol) is irrelevant to the IP family. NetworkManager allows
+            # you to confiugre `ipv[46].dns-*` options (separated by IP family), it is
+            # a misdesign in the API. The role implemented it correctly, but when these
+            # options need to be set, it should configure both IPv4 and IPv6 setting in
+            # NetworkManager. And NetworkManager disallows to configure
+            # `ipv4.dns-search` setting when `ipv4.method` is disabled.
```

Comment:
This comment does not seem to belong, here, IMHO. It is already mentioned in the commit message. Also, it is too generic for the specific code.

Reply:
How about 
```
# NetworkManager only allows to configure `ipv4.dns-search` setting when
# IPv4 is enabled
```

Reply:
I changed into:
```
# NetworkManager only allows to configure `ipv4.dns-search` setting when
# IPv4 is enabled
```